api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,126 @@ message JwtHeader {
   string value_prefix = 2;
 }
 
+// This message specifies a Jwt requirement.
+message JwtRequirement {
+  // Specify the required issuer for the JwtProvider.
+  string issuer = 1;
+
+  // This audiences field overrides the one in JwtProvider if not empty.
+  repeated string audiences = 2;
+}
+
+// This message specifies a per route based Jwt config.
+// Route.per_filter_config for jwt_authn filter uses this message.
+message PerRouteConfig {
+  // Specify list of required JwtProviders. If list is empty,  Jwt is NOT required.
+  // If any of required JwtProviders is missing, or its verification fails, the request will
+  // be rejected.
+  // If a Jwt is not required, but it is in the providers list, it will be verified.
+  // But it will not fail the request if its verification fails.
+  repeated JwtRequirement requires = 1;
+}
+
+// This message specifies a Jwt requirement for a specific Route condition.
+// For example:
+//
+// .. code-block:: yaml
+//
+//    - match:
+//         prefix: "/healthz"
+//    - match:
+//         prefix: "/"
+//      - requires:
+//          issuer: jwt_issuer1
+//      - requires:
+//          issuer: jwt_issuer2
+//          audiences: [audience1, audience2]
+//
```

Comment:
I think the YAML examples above are a bit incorrect ("requires" field, dashes should be on issuer lines I think), could you double check @qiwzhang?